tests/robot_crawler_test.php

```diff
@@ -41,18 +41,48 @@ protected function setUp() {
 
     }
 
-    public function test_absolute_urls() {
-        $this->resetAfterTest(true);
-
-        $base = "http://test.com/sub/";
-
-        $this->assertNotEquals("http://test.com/file.php",     $this->robot->absolute_url($base, '/file.php'         ));
-        $this->assertEquals("http://test.com/sub/file.php", $this->robot->absolute_url($base, 'file.php'          ));
-        $this->assertEquals("http://test.com/file.php",     $this->robot->absolute_url($base, '../file.php'       ));
-        $this->assertEquals("http://test.com/sib/file.php", $this->robot->absolute_url($base, '../sib/file.php'   ));
-        $this->assertEquals("mailto:me@test.com",           $this->robot->absolute_url($base, 'mailto:me@test.com'));
-
+    public function provider() {
+        return array(
+            array(
+                'base' => 'http://test.com/sub/',
+                'links' => array(
+                    'mailto:me@test.com' => 'mailto:me@test.com',
+                    '/file.php' => 'http://test.com/sub/file.php',
+                    'file.php' => 'http://test.com/sub/file.php',
+                    '../sub2/file.php' => 'http://test.com/sub2/file.php',
+                    'http://elsewhere.com/path/' => 'http://elsewhere.com/path/'
+                )
+            ),
+            array(
+                'base' => 'http://test.com/sub1/sub2/',
+                'links' => array(
+                    'mailto:me@test.com' => 'mailto:me@test.com',
+                    '../../file.php' => 'http://test.com/file.php',
+                    'file.php' => 'http://test.com/sub1/sub2/file.php',
+                    '../sub3/file.php' => 'http://test.com/sub1/sub3/file.php',
+                    'http://elsewhere.com/path/' => 'http://elsewhere.com/path/'
+                )
+            ),
+            array(
+                'base' => 'http://test.com/sub1/sub2/$%^/../../../',
+                'links' => array(
+                    'mailto:me@test.com' => 'mailto:me@test.com',
+                    '/file.php' => 'http://test.com/file.php',
+                    '/sub3/sub4//$%^/../../../file.php' => 'http://test.com/file.php',
+                    'http://elsewhere.com/path/' => 'http://elsewhere.com/path/'
+                )
+            )
+        );
     }
 
+    /**
+     * @dataProvider provider
+     */
```

Comment:
this is missing phpdoc stuff, please run the checkers locally you should be picking this up before it gets to travis